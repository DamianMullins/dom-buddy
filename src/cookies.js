/**
 * @overview cookie utilities.
 *
 * @module cookies
 */

/**
 * Get the cookie content.
 *
 * @returns {string}
 */

export const getCookieValue = () => document.cookie;

/**
 * Set the cookie content.
 *
 */

export const setCookieValue = value => (document.cookie = value);
