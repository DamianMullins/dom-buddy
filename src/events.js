/* eslint-disable import/prefer-default-export */
/**
 * @overview Event utilities.
 *
 * @module events
 */


/**
 * Creates and dispatches a DOM event
 *
 * @param {Node} [element=document]
 * @param {string} type
 * @param {object} detail
 *
 * @example
 * dispatchEvent(window, 'onEvent');
 *
 * @see {@link https://developer.mozilla.org/en-US/docs/Web/API/Event|Event - MDN}
 */
export const dispatchEvent = (element = document, type, detail) => {
    let event;

    if (detail) {
        event = new CustomEvent(type, { detail });
    } else {
        event = new Event(type);
    }

    element.dispatchEvent(event);

    return event;
};
