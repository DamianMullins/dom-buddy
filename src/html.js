/**
 * @overview HTML utilities.
 *
 * @module html
 */


/**
 * Get the HTML content of the body.
 *
 * @returns {string}
 */
export const getHtml = (element = document.body) => element.innerHTML;

/**
 * Sets the inner HTML content of an element.
 *
 * @param {string} element HTML content will be set for this element.
 * @param {string} html The HTML content to set.
 */
export const setHtml = (element, html) => {
    element.innerHTML = html;
};

/**
 * Sets the inner HTML content of the document body.
 *
 * @param {string} html The HTML content to set.
 */
export const setBodyHtml = html => {
    setHtml(document.body, html);
};
