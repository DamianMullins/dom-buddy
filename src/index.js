export { getHtml, getBodyHtml, setHtml, setBodyHtml } from './html';

export { dispatchEvent } from './events';
