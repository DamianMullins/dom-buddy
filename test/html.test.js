import { getHtml, setHtml, setBodyHtml } from '../src/html';


describe('getHtml()', () => {

    it('returns html from the document body by default', () => {
        // Arrange
        const html = '<h1>Heading</h1>';
        document.body.innerHTML = html;

        // Act
        const result = getHtml();

        // Assert
        expect(result).toBe(html);
    });

    it('returns html from specified element', () => {
        // Arrange
        const headingText = 'Heading';
        const html = `<main>
                          <h1>${headingText}</h1>
                      </main>`;
        document.body.innerHTML = html;
        const heading = document.querySelector('h1');

        // Act
        const result = getHtml(heading);

        // Assert
        expect(result).toBe(headingText);
    });

});

describe('setHtml()', () => {

    it('sets the document body html', () => {
        // Arrange
        const html = '<h1>Heading</h1>';

        // Act
        setHtml(document.body, html);

        // Assert
        expect(document.body.innerHTML).toMatchSnapshot();
    });

    it('sets element html', () => {
        // Arrange
        const headingText = 'Heading';
        const html = `<main>
                          <h1>Welcome</h1>
                      </main>`;
        document.body.innerHTML = html;
        const heading = document.querySelector('h1');

        // Act
        setHtml(heading, headingText);

        // Assert
        expect(document.body.innerHTML).toMatchSnapshot();
    });

});

describe('setBodyHtml()', () => {

    it('sets the document body html', () => {
        // Arrange
        const html = '<h1>Heading</h1>';

        // Act
        setBodyHtml(html);

        // Assert
        expect(document.body.innerHTML).toMatchSnapshot();
    });

});
