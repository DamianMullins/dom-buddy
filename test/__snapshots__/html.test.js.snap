// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setBodyHtml() sets the document body html 1`] = `"<h1>Heading</h1>"`;

exports[`setHtml() sets element html 1`] = `
"<main>
                          <h1>Heading</h1>
                      </main>"
`;

exports[`setHtml() sets the document body html 1`] = `"<h1>Heading</h1>"`;
